### PR TITLE
make constraint-aware planning more reliable in move_group_interface_tutorial.cpp

### PR DESCRIPTION
**Pull request by mcevoyandy**
Orientation constraint example only works maybe 10% of the time. Seems to work more consistently with these values. Is there a way to check you'll always get a solution? I just moved the start closer to the goal so there's no guarantee these are good points either. When the tutorial solution failed it wasn't immediately clear why. Confusing for someone trying to learn.

### CODE REVIEW

**Comment by davetcoleman** [comment]
Did you try instead to increase the planning time? Something like ``my_plan.planning_time_ = 60;`` Seems like a similar solution that would be less invasive. There is a ``setPlanningTime()`` for this also. By default its 5 in the tutorial, I think.

**Comment by mcevoyandy** [comment]
So this seems to help some. Just ran a loop and got 22/30 to complete.

**Comment by davetcoleman** [comment]
That's still not very impressive. Does the tutorial show how to change which planner to use? The default, KPIECE, is probably the problem. You want RRTConnect instead. See ongoing discussion: https://github.com/ros-planning/moveit/issues/197

**Comment by davetcoleman** [comment]
Can you update this PR to use ``setPlanningTime()`` instead?

**Comment by mcevoyandy** [comment]
Even though it still doesn't work that well?


Andy McEvoy

Research Assistant
Correll Lab
Department of Computer Science
University of Colorado at Boulder

(832) 439 - 6150

**Comment by davetcoleman** [comment]
You reported an increase in solved problems of 63%, so yes, I'd say its a big improvement.

File: doc/pr2_tutorials/planning/src/move_group_interface_tutorial.cpp [hunk 1]
```diff
@@ -177,7 +177,9 @@ int main(int argc, char **argv)
   // We will reuse the old goal that we had and plan to it.
   // Note that this will only work if the current state already 
   // satisfies the path constraints. So, we need to set the start
-  // state to a new pose. 
+  // state to a new pose. We also increase the default planning time from 5s to 60s.
+  
+  my_plan.planning_time_ = 60;
```
Comment (davetcoleman) [inline doc/pr2_tutorials/planning/src/move_group_interface_tutorial.cpp]: From your testing was 60s sometimes used? That seems really really high. How about 30s so MoveIt! doesn't look as bad?

**Comment by davetcoleman** [comment]
You need to rebase with ``kinetic-devel`` due to merge conflicts

File: doc/pr2_tutorials/planning/src/move_group_interface_tutorial.cpp [hunk 2]
```diff
@@ -177,7 +177,9 @@ int main(int argc, char **argv)
   // We will reuse the old goal that we had and plan to it.
   // Note that this will only work if the current state already
   // satisfies the path constraints. So, we need to set the start
-  // state to a new pose.
+  // state to a new pose. We also increase the default planning time from 5s to 10s.
+  
+  my_plan.planning_time_ = 10;
```
Comment (v4hn) [inline doc/pr2_tutorials/planning/src/move_group_interface_tutorial.cpp]: I'm pretty sure this does not work at all.
The `plan` method does not read this value, but overrides it with the [*actual* time used by the planner](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp#L705)

You should use `group.setPlanningTime(10.0)` instead.

**Comment by mcevoyandy** [comment]
I'm not sure, I added that line on dave's recommendation and the tutorial actually worked. I'm going to close this since increasing the time still doesn't give you a beginner tutorial that actually works. Right now there's no explanation or cue to the person taking the tutorial that this part might not work....

**Comment by v4hn** [comment]
> Right now there's no explanation or cue to the person taking the tutorial that this part might not work....

Feel free to open a request that adds only this comment. This is definitely a good idea!

**Comment by davetcoleman** [comment]
@mcevoyandy see https://github.com/ros-planning/moveit/issues/337

**Pull request closed without merging**